lib/pypechain/pypechain/utilities/abi.py

```diff
@@ -0,0 +1,86 @@
+"""Utilities for solidity contract ABIs."""
+from __future__ import annotations
+
+import json
+from typing import List, NamedTuple, Optional, TypeGuard
+
+from web3.types import ABIEvent, ABIFunction
+
+
+class Input(NamedTuple):
+    """An input of a function or event."""
+
+    internalType: str
+    name: str
+    type: str
+    indexed: Optional[bool] = None
+
+
+class Output(NamedTuple):
+    """An output of a function or event."""
+
+    internalType: str
+    internalType: str
+    name: str
+    type: str
+
+
+class AbiItem(NamedTuple):
+    """An item of an ABI, can be an event, function or struct."""
+
+    type: str
+    inputs: List[Input]
+    stateMutability: Optional[str] = None
+    anonymous: Optional[bool] = None
+    name: Optional[str] = None
+    outputs: Optional[List[Output]] = None
+
+
+class AbiJson(NamedTuple):
+    """A JSON representation of a solidity contract's Application Boundary Interface."""
+
+    abi: List[AbiItem]
+
+
+def load_abi(abi_path: str) -> AbiJson:
```

Comment:
we have a version of this in ethpy. Not sure if we want pypechain to depend on ethpy, though? I think eventually we'd expect it to be the other way around, ethpy depends on pypechain. So maybe we can just bake that in now (or in a follow-up PR)? Where ethpy uses this func.

Reply:
no need to act on this now, just musing

Reply:
yeah i'll duplicate for now, if there's more than 3 overlapping i'll make the change.